### ISSUE TITLE
Problems with linked editing ranges through Clojure LSP

### ISSUE THREAD

**Issue by edahlseng** [zed-industries/zed#20002]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

I'm running into a couple of issues with the linked editing ranges feature through the Clojure LSP:
1. Linked editing ranges are being triggered across spaces
2. Linked editing ranges are editing symbols when I wouldn't expect them too

Problem 1 above appears to be a bug; I don't know if problem 2 is a bug or an unfortunate side-effect of the linked editing ranges feature. I've attached a screenshot exhibiting the issues below.

I do not know if these issues are with Zed, with Zed's Clojure extension, or the Clojure LSP. Is there a way to turn off a specific LSP feature? If so, I could at least turn off the linked editing ranges feature as a workaround.

https://github.com/user-attachments/assets/10d07b08-e250-4558-8813-8d6be8267d65

(carrots are a fruit, right?! 😄 )

### Environment

Zed: v0.158.2 (Zed)
OS: macOS 14.6.1
Memory: 16 GiB
Architecture: x86_64

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary><pre>

</pre></details>

**Comment by osiewicz** [comment]
> I do not know if these issues are with Zed, with Zed's Clojure extension, or the Clojure LSP. Is there a way to turn off a specific LSP feature? If so, I could at least turn off the linked editing ranges feature as a workaround.

Yes, there's a `linked_edits` property on a language that you can set to true or false:
```
"languages": {
  "Clojure": {
    "linked_edits": false
  }
}
```
I'll take a look at your repro in a bit. Thanks for making it so concise.

**Comment by kantuni** [comment]
[Elm extension](https://github.com/zed-industries/zed/issues/21843) had the same issue. I'm guessing the issue is coming from the LSP implementation, but I might be wrong.

@edahlseng completely unrelated, what theme are you using? Looks gorgeous!

**Comment by paulbutcher** [comment]
@edahlseng This should now be fixed with the most recent version of clojure-lsp (see https://github.com/clojure-lsp/clojure-lsp/pull/1947 and https://github.com/clojure-lsp/clojure-lsp/pull/1948).

Can you confirm?
